diskqueue.go

```diff
@@ -395,6 +401,223 @@ func (d *diskQueue) readOne() ([]byte, error) {
 	return readBuf, nil
 }
 
+// get the size of the metaData file or its max possible size
+func (d *diskQueue) metaDataFileSize() int64 {
+	var err error
+	var metaDataFile *os.File
+
+	metaDataFile, err = os.OpenFile(d.metaDataFileName(), os.O_RDONLY, 0600)
+
+	var metaDataFileSize int64
+	if err == nil {
+		var stat os.FileInfo
+
+		stat, err = metaDataFile.Stat()
+		if err == nil {
+			metaDataFileSize = stat.Size()
+		}
+	}
+	if err != nil {
+		// use max file size (8 int64 fields)
+		metaDataFileSize = 64
+	}
+
+	return metaDataFileSize
+}
+
+func (d *diskQueue) removeReadFile() error {
+	var err error
+
+	if d.readFileNum == d.writeFileNum {
+		d.skipToNextRWFile()
+		return nil
+	}
+
+	if d.readFile == nil {
+		curFileName := d.fileName(d.readFileNum)
+
+		d.readFile, err = os.OpenFile(curFileName, os.O_RDONLY, 0600)
+		if err != nil {
+			return err
+		}
+
+		d.reader = bufio.NewReader(d.readFile)
+	}
+
+	closeReadFile := func() {
+		d.readFile.Close()
+		d.readFile = nil
+	}
+	defer closeReadFile()
+
+	// read total messages number at the end of the file
+	_, err = d.readFile.Seek(-numFileMsgBytes, 2)
+	if err != nil {
+		return err
+	}
+
+	var totalMessages int64
+	err = binary.Read(d.reader, binary.BigEndian, &totalMessages)
+	if err != nil {
+		return err
+	}
+
+	// update depth with the remaining number of messages
+	d.depth -= totalMessages - d.readMessages
+
+	// we have not finished reading this file
+	if d.readFileNum == d.nextReadFileNum {
+		d.nextReadFileNum++
+		d.nextReadPos = 0
+	}
+
+	d.moveToNextReadFile()
+
+	return nil
+}
+
+// walk through all of the files in the DiskQueue directory
+func (d *diskQueue) walkDiskQueueDir(fn func(fs.DirEntry) error) error {
+	// the directory containing DiskQueue files
+	var mainDir string
+	if d.dataPath == "/" {
+		mainDir = d.dataPath
+	} else {
+		pathArray := strings.Split(d.dataPath, "/")
+		mainDir = pathArray[len(pathArray)-1]
+	}
+
+	walkDiskQueueDir := func(pathStr string, dirEntry fs.DirEntry, err error) error {
+		if dirEntry.Name() == mainDir {
+			// we want to see the contents of the directory DiskQueue writes and reads in
+			return nil
+		}
+
+		if dirEntry.IsDir() {
+			// if the entry is a directory, skip it
+			return fs.SkipDir
+		}
+
+		if err != nil {
+			return err
+		}
+
+		return fn(dirEntry)
+	}
+
+	err := filepath.WalkDir(d.dataPath, walkDiskQueueDir)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *diskQueue) getAllBadFileInfo() ([]fs.FileInfo, error) {
+	var badFileInfos []fs.FileInfo
+
+	getAllBadFileInfo := func(dirEntry fs.DirEntry) error {
+		// only accept "bad" files created by this DiskQueue object
+		regExp, _ := regexp.Compile(`^` + d.name + `.diskqueue.\d\d\d\d\d\d.dat.bad$`)
```

Comment:
Nice catch! It is more of a constant, so I will move it to a global variable. With regards to matching the tail of the string with `.dat.bad`, I was wondering if it would be more accurate to check the whole file name in case another DiskQueue object is also writing to the same directory or the directory holds `.dat.bad` files not related to the current DiskQueue object.